workshops/socialauth.py

```diff
@@ -0,0 +1,76 @@
+from django.http import HttpResponseForbidden
+
+from django.http import HttpResponse
+
+from selectable.decorators import results_decorator
+
+from django.contrib.auth.mixins import UserPassesTestMixin
+
+from django.contrib.auth.decorators import user_passes_test
+
+from django.contrib.auth.decorators import login_required
+
+from django.core.urlresolvers import reverse
+
+from django.shortcuts import redirect, render
+
+from workshops.models import Person
+
+
+class NoUserAssociatedWithGithubAccount(Exception):
+    pass
+
+
+def abort_if_no_user(user, *args, **kwargs):
+    if user is None:
+        raise NoUserAssociatedWithGithubAccount
+
+
+class NoUserAssociatedWithGithubAccountMiddleware(object):
+    def process_exception(self, request, exception):
+        if isinstance(exception, NoUserAssociatedWithGithubAccount):
+            return redirect(reverse('no_github_user'))
+
+
+def find_user_or_abort(details=None, **kwargs):
+    username = details['username']
+    try:
+        user = Person.objects.get(github=username)
+        return {'user': user}
+    except Person.DoesNotExist:
+        raise NoUserAssociatedWithGithubAccount
+
+
+def debug(*args, **kwargs):
+    print('---')
+    for k, v in kwargs.items():
+        print('{} -> {}'.format(k, v))
+
+
+def is_admin(user):
+    if user:
+        return user.groups.filter(name='Admins').exists() or user.is_staff or user.is_superuser
+    return False
+
+
+def no_github_account(request):
+    return render(request, "account/no_github_user.html", {"title": "Log in"})
+
+
+def only_for_admins(f):
+    f = user_passes_test(is_admin)(f)
+    return f
+
+
+@results_decorator
+def selectable_only_for_admins(request):
+    user = getattr(request, 'user', None)
+    if user is None or not user.is_authenticated():
+        return HttpResponse(status=401)  # Unauthorized
+    elif not is_admin(user):
+        return HttpResponseForbidden()
```

Comment:
When `None` is returned, the response is what the view returns. I've added explicit `else: return None`.